dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
+      spells_data_values.find { |data| data['abbrev'].casecmp?(abbrev_to_find) }
+    end
+
+    # Given a hash for a single spell definition,
+    # populates unset properties with base spells data.
+    #
+    # For example:
+    # if spell data has:
+    #   { name: 'Death Monkey', abbrev: 'DM', mana: 5, skill: 'Targeted Magic' }
+    #
+    # and your spell setting in your yaml has:
+    #   { abbrev: 'DM', mana: 8 }
+    #
+    # then this method returns the enriched spell configuration as:
+    #   { name: 'Death Monkey', abbrev: 'DM', mana: 8, skill: 'Targeted Magic' }
+    #
+    enrich_spells_with_data_block = lambda do |spell_setting|
+      return spell_setting unless spell_setting.is_a?(Hash)
+      spell_data = get_spell_data_by_name.call(spell_setting['name']) ||
+                   get_spell_data_by_abbrev.call(spell_setting['abbrev'])
+      (spell_data || {}).merge(spell_setting)
+    end
+
+    # Given a hash of spell names to spell definitions,
+    # populates the `name:` property on the spell definition if it's missing.
+    #
+    # Because the the map keys are themselves the spell names,
+    # some players don't repeat the `name:` or `abbrev:` property
+    # in the spell setting. However, we need that information
+    # in order to enrich the spell setting with base data.
+    #
+    # For example:
+    # if spells map has:
+    #   { 'Devour': { 'use_auto_mana': true }, 'wound_level_threshold': 2 }
+    #
+    # then this method returns the enriched spell configuration as:
+    #   { 'Devour': { 'name': 'Devour', 'use_auto_mana': true }, 'wound_level_threshold': 2 }
+    #
+    enrich_spells_with_name_block = lambda do |spells_map|
+      spells_map.each do |spell_name, spell_setting|
+        # Sometimes these maps support keys other than spell names,
+        # such as numeric thresholds or other related script settings.
+        # Check if this setting is a hash to avoid "no implicit conversion" errors
+        # when we try to access the `name` property.
+        if spell_setting.is_a?(Hash)
+          spell_setting['name'] ||= spell_name
         end
       end
     end
 
-    #get includes from main file and store in a separate variable, so you can't load
-    #includes from includes: prevents circular logic/infinite loops
-    include_list = settings[:include]
-
-    include_list.each do |include_file|
-      filepath = "./scripts/profiles/include-#{include_file}.yaml"
-      if File.exists?(filepath)
-        settings.merge!(safe_load_yaml(filepath))
+    # Waggle sets are maps of arbitrary names to map of spell names to map of spell properties.
+    # For example:
+    #   { 'burgle': { 'Refractive Field': { abbrev: 'RF' ... }, 'Death Monkey': { ... } } }
+    # Iterate each waggle's list of spells to enrich them.
+    # Enrich each's spell definition by merging base spell data with spell settings.
+    settings.waggle_sets.transform_values! do |spells_map|
```

Comment:
Methinks we have too many ways to define spells... not part of this PR, and super difficult to change without an upconverter for settings. Which is an interesting idea now that I mention it...

Reply:
I looked into settings converters back when I worked with making an in game script that would help you generate your settings file by watching you interact with your gear. The biggest issue is just that it likely wipes out all comments and doesn't respect ordering of the given yaml file, so they'll look kind of machine generated after processing.

Reply:
Oof, yeah, good point.

Reply:
Most transforms were largely meant to be temporary, It might be time to start cutting some old formats off, or if we want to support certain quality of life short hands let the consuming scripts do that work instead of the settings file.

Reply:
Wait, can we only upconvert a specific setting (e.g. `offensive_spells`)? That might make things easier. Or make the conversion a separate script and ask people to run it and check the result.

Reply:
So the latter, certainly, the former is basically where I stopped since you'd end up having to do manual text parsing. You'd basically need to read the existing yaml, convert, export that field to text, paste that text over the old block. But that means you need to be able to read and understand a yaml file, although we could do some crude approximations looking for the key and then snipping everything indented under it.

Reply:
> Methinks we have too many ways to define spells...

haha, I came to the same conclusion after working on this!